pkg/build/util/generate.go

```diff
@@ -30,3 +32,44 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
```

Comment:
I don't like this `switch`, I'd propose changing it into:
```
	switch build.Parameters.Strategy.Type {
	case api.DockerBuildStrategyType:
		...
	case api.STIBuildStrategyType:
		...
	case api.CustomBuildStrategyType:
		...
	}
```
The other thing to consider is whether checking if `build.Parameters.Strategy.DockerStrategy != nil` for `DockerBuildStrategyType`, but that's a minor.